test/sparse/CS/correctness.chpl

```diff
@@ -0,0 +1,164 @@
+use LayoutCS;
+use Random;
+
+/*
+  Tests used for development of LayoutCS
+
+  Matrix originated from Colorado State's useful example:
+
+    http://www.cs.colostate.edu/~mcrob/toolbox/c++/sparseMatrix/sparse_matrix_compression.html
+ */
+
+proc main() {
+  var D = {1..6, 1..6};
+
+  var csrDom: sparse subdomain(D) dmapped CS(row=true),
+      cscDom: sparse subdomain(D) dmapped CS(row=false);
+
+  var indices = [(1,1), (1,2), (1,3), (1,4),
+                 (2,2), (2,3),
+                 (3,3), (3,4), (3,5), (3,6),
+                 (4,4), (4,5),
+                 (5,6),
+                 (6,6)];
+
+  writeln('DSI');
+  writeln('===');
+
+  // dsiBulkAdd with domain
+  csrDom += D;
+  cscDom += D;
+
+  assert(csrDom.size == D.size);
+  assert(cscDom.size == D.size);
```

Comment:
I agree.  It would surprise me if the default implementation of `==` on sparse worked (but I'm saying that without taking a look to check).  It won't be particularly efficient, but I think that's acceptable.